### PR TITLE
plugin: 'kotlin-android-extensions' deprecrated

### PR DESCRIPTION
**Pull request by tobiasKaminsky**
use viewbinding
see: https://developer.android.com/topic/libraries/view-binding/migration

This heavily fails, it is mainly a reminder to do it at some point.

Signed-off-by: tobiasKaminsky <tobias@kaminsky.me>

### Testing 
Writing tests is very important. Please try to write some tests for your PR. 
If you need help, please do not hesitate to ask in this PR for help.

[unit tests](https://github.com/nextcloud/android/blob/master/CONTRIBUTING.md#unit-tests)
[instrumented tests](https://github.com/nextcloud/android/blob/master/CONTRIBUTING.md#instrumented-tests)
[UI tests](https://github.com/nextcloud/android/blob/master/CONTRIBUTING.md#ui-tests)

- [x] Tests written, or not not needed

### CODE REVIEW

File: src/main/java/com/nextcloud/client/errorhandling/ShowErrorActivity.kt [hunk 1]
```diff
@@ -28,27 +28,30 @@ import android.widget.Toast
...
-        setContentView(R.layout.activity_show_error)
 
-        text_view_error.text = intent.getStringExtra(EXTRA_ERROR_TEXT)
+        binding = ActivityShowErrorBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        binding.textViewError.text = intent.getStringExtra(EXTRA_ERROR_TEXT)
 
-        setSupportActionBar(toolbar)
+        setSupportActionBar(findViewById(R.id.toolbar))
```
Comment (AndyScherzinger) [inline src/main/java/com/nextcloud/client/errorhandling/ShowErrorActivity.kt:50]: should be done via the binding (needs an id on the include tag to work)

**Review by AndyScherzinger — APPROVED** [review]
just a minor ocmment about the use of findById

File: src/main/java/com/nextcloud/client/errorhandling/ShowErrorActivity.kt (hunk 1 above)
Comment (tobiasKaminsky) [reply src/main/java/com/nextcloud/client/errorhandling/ShowErrorActivity.kt:50]: I tried and failed real hard.
Feel free to give it a try. All my attempts failed to build or had then a NPE crash…

Comment (AndyScherzinger) [reply src/main/java/com/nextcloud/client/errorhandling/ShowErrorActivity.kt]: Can you test? Added a commit: https://github.com/nextcloud/android/pull/7620/commits/cbaa885263e851e13670014a98cb82d40556c403

File: src/main/java/com/nextcloud/client/errorhandling/ShowErrorActivity.kt [hunk 2]
```diff
@@ -47,7 +47,7 @@ class ShowErrorActivity : AppCompatActivity() {
 
         binding.textViewError.text = intent.getStringExtra(EXTRA_ERROR_TEXT)
 
-        setSupportActionBar(findViewById(R.id.toolbar))
+        setSupportActionBar(binding.toolbarInclude.toolbar)
```
Comment (tobiasKaminsky) [inline src/main/java/com/nextcloud/client/errorhandling/ShowErrorActivity.kt:50]: Nice! :+1:

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky wil this PR have to wait until all butterknife parts have been removed before we can merge this?

**Comment by tobiasKaminsky** [comment]
> @tobiasKaminsky wil this PR have to wait until all butterknife parts have been removed before we can merge this?

This is independent from Butterknife.
It is about this special Kotlin usage where one could directly access UI elements by name in code.
This is deprecated and replaced by view bindings.

**Comment by AndyScherzinger** [comment]
> It is about this special Kotlin usage where one could directly access UI elements by name in code.
This is deprecated and replaced by view bindings.

It is is fine since it compiles and can be merged?

**Comment by tobiasKaminsky** [comment]
> It is is fine since it compiles and can be merged?

Yes :+1:

**Pull request merged**